### ISSUE TITLE
Failed to Capture Recursion Error

### ISSUE THREAD

**Issue by oeway** [pyodide/pyodide#951]
Related to https://github.com/iodide-project/pyodide/issues/821, I noticed that we cannot capture the recursion error.

Here is what I got in https://pyodide-cdn2.iodide.io/v0.16.1/full/console.html:
```py
>>> def f(): return 1 + f()
... 
>>> try: f()
... except: print('error')
... 
RangeError: Maximum call stack size exceeded  <---------this is a js error
>>>
``` 
The even more wired thing is, after hitting this error, I cannot run `import micropip`:
```py
>>> print('134')
134
>>> import micropip
Error: Traceback (most recent call last):
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 70, in eval_code
    eval(compile(mod, "<exec>", mode="exec"), ns, ns)
  File "<exec>", line 1, in <module>
  File "/lib/python3.8/site-packages/micropip.py", line 18, in <module>
    import hashlib
  File "/lib/python3.8/hashlib.py", line 167, in <module>
    import _hashlib
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 914, in _find_spec
  File "<frozen importlib._bootstrap_external>", line 1342, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1309, in _get_spec
RecursionError: maximum recursion depth exceeded in __instancecheck__
```

In a native python3, I can get the correct output:
```py
>>> def f(): return 1 + f()
... 
>>> try: f()
... except: print('error')
... 
error
```

Go back to the pyodide console, we can capture the error in JS, for example, run the following code and I can see the error captured by JS. However, it still makes python stuck into a state that no additional import can be made.
```py
>>> import js
>>> js.eval('try{pyodide.runPython("def f(): return 1 + f()\\nf()")}catch(e){console.error("error!")}')
>>> import micropip
Error: Traceback (most recent call last):
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 70, in eval_code
    eval(compile(mod, "<exec>", mode="exec"), ns, ns)
  File "<exec>", line 1, in <module>
  File "/lib/python3.8/site-packages/micropip.py", line 18, in <module>
    import hashlib
  File "/lib/python3.8/hashlib.py", line 167, in <module>
    import _hashlib
  File "<frozen importlib._bootstrap>", line 988, in _find_and_load
  File "<frozen importlib._bootstrap>", line 148, in __enter__
  File "<frozen importlib._bootstrap>", line 174, in _get_module_lock
  File "<frozen importlib._bootstrap>", line 59, in __init__
RecursionError: maximum recursion depth exceeded while calling a Python object
>>> import js
>>> js.pyodide.runPython('123')
123
>>> js.pyodide.runPython('import random')
Error: Traceback (most recent call last):
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 70, in eval_code
    eval(compile(mod, "<exec>", mode="exec"), ns, ns)
  File "<exec>", line 1, in <module>
RecursionError: maximum recursion depth exceeded during compilation
```

I am trying to fix https://github.com/iodide-project/pyodide/issues/917 by capture the recursion error. It basically occurs when I try to run `importjs; js.pyodide.runPython("import skimage")` in a web loop.

**Comment by hoodmane** [comment]
The recursion limit is set too high. I think the problem is that `fixRecursionLimit` assumes that one python stack entry is no larger than 50 empty js function calls. I think in practice it's closer to 70-80 native function calls.

One thing to note is that Python does not measure how much stack space it's consuming, it only counts how many nested python function calls have occurred. I think this may mean that at the same recursion depth, the amount of stack space consumed can vary quite a lot. The recursion limit has to be set conservatively so that even when the function calls take up a large amount of stack space they still never cause a low level stack overflow.

> The even more wired thing is, after hitting this error, I cannot run import micropip:

When you see a `RangeError`, the pyodide runtime has suffered a fatal error. If it were running on an operating system, the operating system would kill the runtime. Try the following on your native python:
```python
def f(): f()
sys.setrecursionlimit(1000000)
f()
```
My OS prints `Segmentation fault` and kills Python. The browser allows the program to continue, but the state of the python VM is likely invalid and the behavior of all further python code executed is undefined. Anyways, catching the range error in javascript is completely wrongheaded unless the handling code kills pyodide and prints "Pyodide has suffered a fatal error!"

**Comment by hoodmane** [comment]
Closed by #1151 which will unambiguously prevent anyone else from trying to do this.
